src/backends/tensorrt/tensorrtlib.cc

```diff
@@ -406,32 +407,28 @@ namespace dd
         _net_mutex); // no concurrent calls since the net is not
                      // re-instantiated
 
-    if (ad.getobj("parameters").getobj("mllib").has("gpuid"))
-      _gpuid = ad.getobj("parameters").getobj("mllib").get("gpuid").get<int>();
+    auto predict_dto = ad.createSharedDTO<DTO::ServicePredict>();
+
+    _gpuid = predict_dto->parameters->mllib->gpuid->_ids[0];
```

Comment:
_ids size can be 0 in the case where the user provides an empty array as gpuid, so we should at least prevent the segfault by using `at()` instead of `[]`

Reply:
as using tensorrt w/o any gpu does not make sense, I will put gpuid to zero if _ids.size() == 0 ; does it seem ok for you ?